felix/bpf/tc/compiler.go

```diff
@@ -121,3 +124,22 @@ func ProgFilename(epType EndpointType, toOrFrom ToOrFromEp, epToHostDrop, fib, d
 		toOrFrom, epTypeShort, hostDropPart, fibPart, dsrPart, logLevel, corePart)
 	return oFileName
 }
+
+const (
+	PinBaseDir = "/sys/fs/bpf/tc/"
+)
+
+func MapPinPath(typ int, name, iface string, hook Hook) string {
+	subDir := "globals"
+	if (typ == unix.BPF_MAP_TYPE_PROG_ARRAY && strings.Contains(name, bpf.JumpMapName())) ||
+		(typ == unix.BPF_MAP_TYPE_PERCPU_ARRAY && strings.Contains(name, bpf.CountersMapName())) {
+		// Remove period in the interface name if any
+		ifName := strings.ReplaceAll(iface, ".", "")
```

Comment:
Hmm, this code was just moved, maybe see if you can dig out the reason and comment it!

Reply:
This is added in this commit: https://github.com/projectcalico/calico/commit/6369bf2599e0d5bfa61c97b4936db5efa0d61190

@sridhartigera do you remember why `.` is removed from interface name?

Reply:
Since we use the ifname in the pin path of the jump map, having a period is not allowed. Hence we remove the period. For this case, ```foobar``` and ```foo.bar```, we need to think of some other scheme.